### PR TITLE
WIP: Add MISRA Support

### LINKED ISSUE #26: Add MISRA Support
Add support in the configuration page, and when invoking cppcheck, of running MISRA checks. 

Unable to distribute MISRA rules texts, under the license which MISRA is distributed, this has to be obtained from them directly and parsed into a cppcheck readable format. This is left as an exercise to the user though.

### PR DESCRIPTION
**Pull request by SJ-Innovation**
Hello, 

This is a WIP PR to inform you that work is being performed on this, before a request to merge is made. 

Closes #26

### CODE REVIEW

**Comment by SJ-Innovation** [comment]
@johnthagen Hi, ive added MISRA support in this PR. Any chance of a review (and merge)? 

Thanks :)

**Comment by johnthagen** [comment]
@firewave As a recent contributor, would you mind also reviewing?

File: src/com/github/johnthagen/cppcheck/Configuration.java [hunk 1]
```diff
@@ -13,16 +13,27 @@
...
   private JTextField cppcheckOptionsField;
+  private JFilePicker cppcheckMisraFilePicker;
   private static final String CPPCHECK_NOTE =
     "Note: C++ projects should leave --language=c++ appended to the cppcheck options to avoid some " +
     "false positives in header files due to the fact that cppcheck implicitly defaults to " +
     "setting --language to \"c\" for .h files.\n\n" +
     "You should not include any --template={} in the options.";
+  private static final String CPPCHECK_MISRA_NOTE =
+    "Using MISRA requires a rule texts file, which can be obtained from MISRA themselves (Their license prohibits distributing the rules texts)\n\n" +
+    "Create a .json file near your CppCheck installation and point to it here\n" +
```
Comment (johnthagen) [inline src/com/github/johnthagen/cppcheck/Configuration.java:24]: Please change all instances of `CppCheck` to `Cppcheck`. Please see: https://github.com/johnthagen/clion-cppcheck/commit/0e9d4fc32957fb603405ece7e61193c7bb2ddb14#commitcomment-38153765

Comment (SJ-Innovation) [reply src/com/github/johnthagen/cppcheck/Configuration.java]: Changed as requested.

**Pull request merged**